ibu-imager/postpivot/postpivot.go

```diff
@@ -0,0 +1,255 @@
+		authFile:             authFile,
+		recertContainerImage: recertContainerImage,
+		workingDir:           workingDir,
+		kubeconfig:           kubeconfig,
+	}
+}
+
+func (p *PostPivot) PostPivotConfiguration(ctx context.Context) error {
+	p.log.Info("Reading cluster info")
+	clusterInfo, err := clusterinfo.ReadClusterInfoFromFile(
+		path.Join(p.workingDir, common.ClusterConfigDir, common.ClusterInfoFileName))
+	if err != nil {
+		return fmt.Errorf("failed to get cluster info from %s, err: %w", "", err)
+	}
+
+	p.log.Info("Reading seed info")
+	seedClusterInfo, err := clusterinfo.ReadClusterInfoFromFile(path.Join(p.workingDir, common.SeedManifest))
+	if err != nil {
+		return fmt.Errorf("failed to get seed info from %s, err: %w", "", err)
+	}
+
+	p.log.Info("Create recert configuration file")
+	if err := recert.CreateRecertConfigFile(clusterInfo, seedClusterInfo, path.Join(p.workingDir, common.CertsDir),
+		p.workingDir); err != nil {
+		return err
+	}
+
+	if err := p.recert(clusterInfo, seedClusterInfo); err != nil {
+		return err
+	}
+
+	client, err := utils.CreateKubeClient(p.scheme, p.kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to create k8s client, err: %w", err)
+	}
+
+	if _, err := p.ops.SystemctlAction("enable", "kubelet", "--now"); err != nil {
+		return err
+	}
+	p.waitForApi(ctx, client)
+	p.approveCsrs(ctx, client)
+
+	if err := p.applyManifests(); err != nil {
+		return err
+	}
+
+	_, err = p.ops.SystemctlAction("disable", "installation-configuration.service")
+	if err != nil {
+		return fmt.Errorf("failed to disable installation-configuration.service, err: %w", err)
+	}
+
+	p.log.Infof("Removing %s", p.workingDir)
+	return os.RemoveAll(p.workingDir)
+}
+
+func (p *PostPivot) recert(clusterInfo, seedClusterInfo *clusterinfo.ClusterInfo) error {
+	doneFile := path.Join(p.workingDir, recertDone)
+	if _, err := os.Stat(doneFile); err == nil {
+		p.log.Infof("Found %s file, skippin recert", doneFile)
+		return nil
+	}
+
+	err := p.ops.RunUnauthenticatedEtcdServer(p.authFile, common.EtcdContainerName)
+	if err != nil {
+		return fmt.Errorf("failed to run etcd, err: %w", err)
+	}
+
+	defer func() {
+		p.log.Info("Killing the unauthenticated etcd server")
+		_, err := p.ops.RunInHostNamespace(
+			"podman", "kill", common.EtcdContainerName)
+		if err != nil {
+			p.log.WithError(err).Errorf("failed to kill %s container.", common.EtcdContainerName)
+		}
+		_, err = p.ops.RunInHostNamespace(
+			"podman", "rm", common.EtcdContainerName)
+		if err != nil {
+			p.log.WithError(err).Errorf("failed to rm %s container.", common.EtcdContainerName)
+		}
+	}()
+
+	if err := p.additionalCommands(clusterInfo, seedClusterInfo, common.EtcdContainerName); err != nil {
+		return err
+	}
+
+	if err := p.ops.RunRecert(p.recertContainerImage, p.authFile, path.Join(p.workingDir, recert.RecertConfigFile),
+		"-v", fmt.Sprintf("%s:%s", p.workingDir, p.workingDir)); err != nil {
+		return err
+	}
+
+	_, err = os.Create(doneFile)
```

Comment:
?

Reply:
if we change err scope in the above comment - https://github.com/openshift-kni/lifecycle-agent/pull/101#discussion_r1400604460

then we need to specify `err` with `:=` here, otherwise you will see this error
```
Unresolved reference 'err'
```